oauth2/oauth2.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oauth2
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+// Config is the client configuration.
+type Config struct {
+	ClientID       string            `yaml:"client_id"`
+	ClientSecret   string            `yaml:"client_secret"`
```

Comment:
This should be a secret.

Reply:
How would I go about making that distinction?

Reply:
by using our secret type. it might mean that we move the code under config to avoid cross dependency

Reply:
Thanks!